main/template/default/auth/courses_categories.php

```diff
@@ -398,6 +435,18 @@ function return_register_button($course, $stok, $code, $search_term)
 }
 
 /**
+ * Display the service list button of a course in the course catalog
+ * @param $id_course
+ */
+function return_service_list_button($id_course)
+{
+    $plugin = BuyCoursesPlugin::create();
```

Comment:
$id_course should be filtered here. For example, intval() or Security::remove_XSS() (I hope it's an integer)